### ISSUE TITLE
Option to load full thumbnails for visible images

### ISSUE THREAD

**Issue by jncraton** [tagspaces/tagspaces#386]
I first want to say that this app is great, and it's something that I've been looking for for a long time. I really appreciate all the hard work that you have put in on this project.

I recently started using this app for tagging photos, and I don't love the way that it handles thumbnails. The current system requires thumbnails to be generated and stored as metadata. I don't love this for the following reasons:

1. It requires the generation of thumbnails. This can certainly be automated, and it sounds like it will be in the future, but it is still annoying.
2. It pollutes my file system with folders filled with thumbnails. This makes sense on a public web server, but seems unnecessary on an individual's pc.

I understand the reason for doing this is related to performance (#83, #309, etc). However, I'm curious if it would be possible to have the option to load the full file as a thumbnail like was possible in previous versions.

One way to increase performance even with full thumbnails would be to only load the thumbnails for files that are currently visible on the screen. I put together a quick hack to show what I'm talking about here:

https://github.com/jncraton/perspectiveGrid/commit/09bd7c03586ec9dd3c3f05cbd5e371c8de9b8499

Basically, this just waits for scroll events and updates the backgrounds for fileTile elements that are currently located within the viewport. I'd love to see a more robust version of this in a release. I'd even be interested in coding this up myself if someone can provide guidance on where this should live in the larger application.

Thanks again for your work on this! I really like this application.

**Comment by jncraton** [comment]
As an optimization, I've tried down sampling images using a canvas element so that the full images never hit the DOM. This appears to provide a nice performance increase when viewing lots of large images. It particularly seems to help FPS when scrolling.

Here's the full change set:

https://github.com/tagspaces/perspectiveGrid/compare/master...jncraton:master

**Comment by uggrock** [comment]
Hi,

thanks for your great contribution. I just tried it on nwjs and electron and it works! We will evaluate it on other platforms like chrome and android and will try to integrate it as soon as possible.

kind regards,
Ilian

**Comment by jncraton** [comment]
Terrific! Thanks for your work on this!

**Comment by uggrock** [comment]
Based on the ideas in your pull request we have now support for the so called live thumbnails in the new 2.1.4 release.

**Comment by jncraton** [comment]
Fantastic! Great work on implementing this so quickly!
